js_modules/dagster-ui/packages/ui-components/src/components/MiddleTruncate.tsx

```diff
@@ -19,34 +39,24 @@ interface Props {
  *
  * When the DOM element resizes, the measurement and calculation steps will occur again.
  */
-export const MiddleTruncate = React.memo(({text, showTitle = true}: Props) => {
+export const MiddleTruncate = memo(({text, showTitle = true}: Props) => {
   // An element that renders the full text into the container, for the purpose of
   // measuring the maximum available/necessary width for our truncated string.
-  const measure = React.useRef<HTMLDivElement>(null);
-  const truncated = React.useRef<HTMLDivElement>(null);
-
-  // Given the target font style and allotted width, calculate the largest possible middle-
-  // truncated string.
-  const calculateTargetStyle = React.useCallback(() => {
-    if (measure.current && truncated.current) {
-      truncated.current.textContent = calculateMiddleTruncatedText(measure.current, text);
+  const measure = useRef<HTMLDivElement>(null);
+  const width = useWidth(measure);
+  const [truncatedText, setTruncatedText] = useState(text);
+
+  useLayoutEffect(() => {
+    if (measure.current) {
+      const font = getComputedStyle(measure.current).font;
+      const result = calculateMiddleTruncatedText({font, width, text});
+      setTruncatedText(result ?? text);
```

Comment:
Setting the text via state instead of directly modifying the `textContent`. I don't recall why it was setting it directly before, but using state seems to be fine.

Reply:
Here's where it was introduced: https://github.com/dagster-io/dagster/pull/26888

I'll see if I can repro that issue, and if so, I'll restore the direct `textContent` setter.

Reply:
I'm not noticing a flicker. I'll go ahead with this.